packages/core/src/ceramic.ts

```diff
@@ -237,93 +290,109 @@ class Ceramic implements CeramicApi {
+    logger.imp(`Starting Ceramic node at version ${packageJson.version} with config: \n${JSON.stringify(config, null, 2)}`)
 
     const anchorService = config.anchorServiceUrl ? new EthereumAnchorService(config) : new InMemoryAnchorService(config as any)
     await anchorService.init()
-    const context: Context = {
-      ipfs,
-      anchorService,
-      logger,
-    }
 
     const networkOptions = await Ceramic._generateNetworkOptions(config, anchorService)
     logger.imp(`Connecting to ceramic network '${networkOptions.name}' using pubsub topic '${networkOptions.pubsubTopic}' with supported anchor chains ['${networkOptions.supportedChains.join("','")}']`)
 
-    const dispatcher = new Dispatcher(ipfs, networkOptions.pubsubTopic, logger, pubsubLogger)
-    await dispatcher.init()
-
-    const pinStoreProperties = {
+    const pinStoreOptions: PinStoreOptions = {
       networkName: networkOptions.name,
       pinsetDirectory: config.stateStoreDirectory,
-      pinningEndpoints: config.ipfsPinningEndpoints,
-      pinningBackends: config.pinningBackends
+      pinningEndpoints: config.ipfsPinningEndpoints
     }
-    const pinStoreFactory = new PinStoreFactory(ipfs, pinStoreProperties)
-    const pinStore = await pinStoreFactory.open()
-    const topology = new IpfsTopology(ipfs, networkOptions.name)
-    const ceramic = new Ceramic(dispatcher, pinStore, context, topology, networkOptions, config.validateDocs, config.docCacheLimit, config.cacheDocCommits)
-    anchorService.ceramic = ceramic
 
-    const keyDidResolver = KeyDidResolver.getResolver()
-    const threeIdResolver = ThreeIdResolver.getResolver(ceramic)
-    ceramic.context.resolver = new Resolver({
-      ...config.didResolver, ...threeIdResolver, ...keyDidResolver,
-    })
+    const params = {
+      cacheDocumentCommits: config.cacheDocCommits ?? true,
+      docCacheLimit: config.docCacheLimit ?? DEFAULT_DOC_CACHE_LIMIT,
+      loggerConfig: config.logger,
+      networkOptions,
+      pinStoreOptions,
+      validateDocs: config.validateDocs ?? true,
+    }
 
-    if (config.didProvider) {
-      await ceramic.setDIDProvider(config.didProvider)
+    const modules = {
+      anchorService,
+      didResolver: config.didResolver,
+      dispatcher: null, // set in Ceramic.create()
+      ipfs: null, // set in Ceramic.create()
+      ipfsTopology: null, // set in Ceramic.create()
+      logger,
+      pinStoreFactory: null, // set in Ceramic.create()
     }
 
+    return [params, modules]
+  }
+
+  /**
+   * Create Ceramic instance
+   * @param ipfs - IPFS instance
+   * @param config - Ceramic configuration
+   */
+  static async create(ipfs: IpfsApi, config: CeramicConfig = {}): Promise<Ceramic> {
+    // todo: Remove ipfs argument, pull logic to start in-process IPFS here and out of CeramicDaemon
+    const [params, modules] = await Ceramic._processConfig(config)
+
+    modules.ipfs = ipfs
```

Comment:
I wasn't able to totally get this review to the place I wanted it today, which is why this PR is a draft, but there's enough moving pieces here I thought it makes sense to get some other eyes on it sooner than later.  The main things I still want to do are:
* add a `setIpfs()` method to the `Ceramic` object that is called after the constructor but before `_init()` and can take care of initializing these ipfs-dependent modules
* Remove the `ipfs` argument from Ceramic.create(), let ceramic internally handle creating the IpfsDaemon or the ipfs http client based on the config, rather than having the CeramicDaemon do it.
* Finally provide the logger instance to the IpfsDaemon and IpfsTopology so they no longer use `console.log` directly.

The above is my plan for tomorrow (today for you already), but if you have any feedback about anything I'm doing here I wanted to give you a chance to give it before I start work in the morning and go further down this path.